realm/mocked_host/run.py

```diff
@@ -0,0 +1,245 @@
+#!/usr/bin/env python3
+
+import argparse
+import uuid
+import shlex
+import json
+import sys
+import subprocess
+import os
+import socket
+import struct
+from typing import List
+from dataclasses import dataclass
+from gpt_image.disk import Disk
+from gpt_image.partition import Partition, PartitionType
+
+APP_ID = uuid.UUID("5d63a211-e8aa-4179-ac22-af7e843a3f43")
+IMAGE_UUID = "0178460a-ffea-4674-8e17-8530758c4c2e"
+DATA_UUID = "259c493c-e5d2-4fd4-a962-7efd45a0bd91"
+
+@dataclass()
+class App:
+    id: str
+    name: str
+    version: str
+    image_registry: str
+    image_part_uuid: str
+    data_part_uuid: str
+
+class MockedWarden():
+    def __init__(self, vsock_port = 1337, guest_cid = 1227, tap_device = "tap100", qemu_serial = "tcp:localhost:1337"):
+        self.vsock_port = vsock_port
+        self.guest_cid = guest_cid
+        self.tap_device = tap_device
+        self.qemu_serial = qemu_serial
+        self.sock = socket.socket(socket.AF_VSOCK, socket.SOCK_STREAM, 0)
+        self.sock.bind((socket.VMADDR_CID_ANY, vsock_port))
+        self.sock.listen(1)
+
+    def prepare_disk(self):
+        if not os.path.isfile("disk.raw"):
+            disk = Disk("disk.raw")
+            disk.create(1024 * 1024 * 1024)
+
+            image_part = Partition("image", 256 * 1024 * 1024, PartitionType.LINUX_FILE_SYSTEM.value, partition_guid=IMAGE_UUID)
+            data_part = Partition("data", 256 * 1024 * 1024, PartitionType.LINUX_FILE_SYSTEM.value, partition_guid=DATA_UUID)
+
+            disk.table.partitions.add(image_part)
+            disk.table.partitions.add(data_part)
+            disk.commit()
+
+    def run_qemu(self):
+        self.qemu = subprocess.Popen(shlex.split(f"""
+ "../tools/qemu/build/qemu-system-aarch64" \
+        -machine virt \
+        -cpu cortex-a57 \
+        -nographic -smp 1 \
+        -kernel /home/m.szaknis/dev/realm-manager/realm/linux/arch/arm64/boot/Image \
```

Comment:
Could you transform this hard coded value into an input variable?

Reply:
Uffff sorry, I didn't catch that.

Reply:
done